index.js

```diff
@@ -1,26 +1,30 @@
 const core = require('@actions/core');
-
-const { execSync } = require('child_process');
+const io = require('@actions/io');
+const exec = require('@actions/exec');
 
 const host = core.getInput('host');
 const port = core.getInput('port');
 const socketPath = core.getInput('socket-path');
 const key = core.getInput('key');
+const lifetimeInSeconds = core.getInput('lifetime_in_seconds');
 
-console.log(`Attempting to create ${socketPath}...`);
-try {
-    execSync(
-        'mkdir -p ~/.ssh && ' +
-        `ssh-keyscan${port ? ` -p ${port}` : ''} "${host}" >> ~/.ssh/known_hosts && ` +
-        `eval $(ssh-agent -a "${socketPath}") && ` +
-        `echo "${key}" | base64 -d | ssh-add -`
-    );
-} catch (e) {
-    console.error(e.message);
-    process.exit(1);
-}
+async function run() {
+    console.log('checking if ~/.ssh is created...');
+    await io.mkdirP('~/.ssh');
+
+    console.log('ssh-keyscan the provided domain...');
+    let portCommand = port ? `-p ${port}` : '';
+    await exec.exec('ssh-key', [portCommand, host, '>> ~/.ssh/known_hosts']);
 
-console.log(`Created ${socketPath}`);
-core.setOutput('socket-path', socketPath);
+    console.log(`Attempting to create if not found. ${socketPath}...`);
+    await exec.exec(`eval $(ssh-agent -a "${socketPath}")`);
```

Comment:
I'm not sure because their docs are sparse and this code is pretty complicated, but I think this will throw if the return is non-zero (which is will be if the socket already exists) https://github.com/actions/toolkit/blob/master/packages/exec/src/toolrunner.ts#L510

I'm assuming we want to catch the stderr and check if it's like `unix_listener: cannot bind to path /tmp/mysock: Address already in use` and keep going.

Reply:
I did an empty catch for now while I test the other parts.